extension/diffheader.js

```diff
@@ -21,6 +21,46 @@ window.diffFileHeader = (() => {
 		};
 	})();
 
+	const maxPxAvailable = () => {
+		// Unfortunately can't cache this value, as it'll change with the browsers zoom level
+		const filenameLeftOffset = $('.diff-toolbar-filename').get(0).getBoundingClientRect().left;
+		const diffStatLeftOffset = $('.diff-toolbar-filename + .right').get(0).getBoundingClientRect().left;
+
+		return diffStatLeftOffset - filenameLeftOffset;
+	};
+
+	const parseFileDetails = filename => {
+		const folderCount = (filename.match(/\//g) || []).length;
+		const [, basename] = (filename.match(/(?:\/)([\w\.-]+)$/) || []);
+		const [, topDir] = (filename.match(/^([\w\.-]+)\//) || []);
+
+		return {
+			folderCount,
+			basename,
+			topDir
+		};
+	};
+
+	const updateFileLabel = val => {
+		const target = $('.diff-toolbar-filename').get(0);
+		target.classList.add('filename-height-check');
+		target.textContent = val;
+
+		const maxPx = maxPxAvailable();
+		const doesOverflow = () => target.getBoundingClientRect().width > maxPx;
+		const {basename, topDir, folderCount} = parseFileDetails(val);
+
+		if (doesOverflow() && folderCount > 1) {
+			target.textContent = `${topDir}/.../${basename}`;
+		}
+
+		if (doesOverflow()) {
+			target.textContent = basename;
+		}
```

Comment:
I had originally done that, but I ended up not doing that because, if the root directory + basename ends up being too long, we'd want to then fall back to just the filename (meaning we have to re-check the overflow). Thoughts?